pkg/flow/delegate.go

```diff
@@ -0,0 +1,125 @@
+package flow
+
+import (
+	"context"
+	"net/http"
+	"path"
+
+	"github.com/gorilla/mux"
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/pkg/cluster"
+	"github.com/grafana/agent/pkg/flow/logging"
+	"github.com/grafana/agent/web/api"
+	"github.com/prometheus/client_golang/prometheus"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type delegate struct {
```

Comment:
this type being called only delegate in the flow package is a bit confusing to me since it is actually a module delegate. The file being delegate.go also implies to me it will contain any delegates for flow. I saw @rfratto 's comment about removing the term delegate in general and this would be impacted as well.

Reply:
Will change.